src/lib/peers/PeerManager.ts

```diff
@@ -146,11 +150,15 @@ class PeerManager {
       this.credentials = grpc.ServerCredentials.createInsecure()
     }
 
-    // Bind server and set address
-    const port = this.server.bind('0.0.0.0:0', this.credentials);
-    const address = new Address('localhost', port.toString())
-    this.server.start();
-    this.localPeerInfo.connect(address)
+    this.server.bindAsync('0.0.0.0:0', this.credentials, (err, boundPort) => {
```

Comment:
BTW, this should come from the CLI parameters or enviroment variable. I would suggest both where the parameters overrides the env variable.

Reply:
We can have things like `PK_X` as our environment variable.

Reply:
oh yes, the port was an environment variable before, I've just forgotten to keep it that way when I changed to `grpc-js`. It is fixed now :+1: `PK_PORT ?? 0`

Reply:
Not just the port, but the address to. So you would have `PK_HOST`.